lib/postgrex/notifications.ex

```diff
@@ -102,19 +102,19 @@ defmodule Postgrex.Notifications do
     ref = Process.monitor(pid)
 
     s = put_in(s.listeners[ref], {channel, pid})
-    s = update_in(s.listener_channels[channel], &((&1 || HashSet.new()) |> HashSet.put(ref)))
+    s = update_in(s.listener_channels[channel], &((&1 || MapSet.new()) |> MapSet.put(ref)))
```

Comment:
@fishcakez if I do something like `s = update_in(s.listener_channels, &((&1 || Map.new()) |> Map.put("channel", ref)))` here. Should this be a list of reference ids? A map would require keys...